### PR TITLE
Fix alert component colors

### LINKED ISSUE #459: Alert component not showing error color
**Describe the bug**

Found while testing #458. I added an `Alert` component to display a message, but it wasn't displayed in red.

From the Vuetify docs, looks like they changed their component. Instead of using `:color="error"`, now they are expecting `:type="error"`.

**Release version(s) and/or repository branch(es) affected?**

`master`

**Steps to reproduce the bug**

Add an alert to the store with `color="error"`. It is displayed as a transparent div.

**Expected behavior**

Div coloured according to the color/type specified, e.g. red for error.

**Screenshots**

Forgot to take a screenshot. Logging issue so I don't forget it :)

**Additional context**

https://v15.vuetifyjs.com/en/components/alerts

**Pull requests welcome!**
This is an Open Source project - please consider contributing a bug fix
yourself (please read `CONTRIBUTING.md` before starting any work though).

### PR DESCRIPTION
**Pull request by kinow**
These changes close #459

**Requirements check-list**
- [x] I have read `CONTRIBUTING.md` and added my name as a Code Contributor.
- [x] Contains logically grouped changes (else tidy your branch by rebase).
- [x] Does not contain off-topic changes (use other PRs for other changes).
- [x] Appropriate tests are included (unit and/or functional).
- [x] No change log entry required (why? e.g. invisible to users).

### CODE REVIEW

**Comment by kinow** [comment]
Preview using `success`, `warning`, and `error` alerts.

image

image

image

**Comment by kinow** [comment]
Was going to tick the box "Does not need a test...", but I think it might be testable actually. Let me give it a try.

File: src/components/core/Alert.vue [hunk 1]
```diff
@@ -19,12 +19,15 @@ along with this program.  If not, see <http://www.gnu.org/licenses/>.
...
       :value="true"
-      :color="alert.getColor()"
+      :type="alert.getColor()"
       :icon="alert.getIcon()"
-      class="ma-0"
+      :class="getColor(alert.getColor())"
       dismissible
+      tile
+      light
+      colored-border
```
Comment (kinow) [inline src/components/core/Alert.vue:28]: `tile` removes the rounded borders.
`light` and `colored-border` to indicate no dark mode (due to upstream bug)

**Comment by kinow** [comment]
What it looks like when you have an alert without the background color in an e2e test:

image

I am not checking whether the error is red, success is green, etc, because I couldn't get the exact values from the Vuetify style dynamically. So instead I just have two constants, one for black and the other for white.

And the tests just go through each type (error, warning, success) and check that the background color is neither black nor white.

**Comment by kinow** [comment]
Spent a good amount of time trying to get the colors; not a good idea unless you want to dig deep into how Cypress and Babel/WebPack/etc work... also had a lot of work trying to write a test using Vuex state, then learned the hard way that Cypress is a bit opinionated on how you must write tests, some of their docs state how they believe tests should be written. Point taken, try to write simple tests, isolated from each other, without sharing states :+1:

**Comment by kinow** [comment]
(coverage went down a bit without the e2e, but now it's back to green :tada: )

File: src/main.js [hunk 2]
```diff
@@ -68,3 +68,9 @@ new Vue({
     })
   }
 }).$mount('#app')
+// e2e tests use the offline mode, so here we expose the Vue.js app so Cypress can access it programmatically
+// e.g. window.app.$store and window.app.$workflowService.
+// Ref: https://www.cypress.io/blog/2017/11/28/testing-vue-web-application-with-vuex-data-store-and-rest-backend/
+if (['test', 'offline'].includes(process.env.NODE_ENV)) {
+  window.app = app
+}
```
Comment (kinow) [inline src/main.js:76]: Same change in #454 . So once either this or that PR is merged, I will fix conflicts in the other PR :+1:

File: src/components/core/Alert.vue [hunk 3]
```diff
@@ -19,12 +19,15 @@ along with this program.  If not, see <http://www.gnu.org/licenses/>.
...
     >
-      <p class="body-1">{{ alert.getText() }}</p>
+      {{ alert.getText() }}
```
Comment (oliver-sanders) [inline src/components/core/Alert.vue:30]: Shouldn't this text be wrapped, at least in a `<span/>`?

File: src/components/core/Alert.vue [hunk 4]
```diff
@@ -19,12 +19,15 @@ along with this program.  If not, see <http://www.gnu.org/licenses/>.
   <div v-if="alert">
     <v-alert
       :value="true"
-      :color="alert.getColor()"
+      :type="alert.getColor()"
       :icon="alert.getIcon()"
-      class="ma-0"
+      :class="getColor(alert.getColor())"
```
Comment (oliver-sanders) [inline src/components/core/Alert.vue:24]: Are these colour classes are built into Vuetify?

File: src/components/core/Alert.vue (hunk 3 above)
Comment (kinow) [reply src/components/core/Alert.vue:30]: The component ends up wrapping the text in some HTML element later I think. I changed after I noticed some padding/margin, and saw that they don't use it in their examples: https://v15.vuetifyjs.com/en/components/alerts

File: src/components/core/Alert.vue (hunk 4 above)
Comment (kinow) [reply src/components/core/Alert.vue:24]: Yup, the material color palette https://v15.vuetifyjs.com/en/framework/colors

Comment (oliver-sanders) [reply src/components/core/Alert.vue:24]: Matches the Cylc logo pretty well!

**Comment by oliver-sanders** [comment]
@datamel quick UI activity, I've done a code review so a functional test and quick sanity check should be sufficient.

**Comment by hjoliver** [comment]
Looks nice to me :tada: ; leaving final review for @datamel though.

**Comment by kinow** [comment]
Thanks @oliver-sanders and @datamel !

**Pull request merged**